dxr/cli/shell.py

```diff
@@ -0,0 +1,49 @@
+from click import command, option
+
+from dxr.app import make_app
+from dxr.cli.utils import config_option
+
+
+def _plain_shell(namespace):
+    """Start a plain python shell."""
+    import code
+    import readline
+    import rlcompleter
+
+    readline.set_completer(rlcompleter.Completer(namespace).complete)
+    readline.parse_and_bind('tab:complete;')
+    code.interact(local=namespace)
+
+
+def _ipython(namespace):
+    """Start IPython."""
+    from IPython import start_ipython
+    start_ipython(argv=[], user_ns=namespace)
+
+
+_shells = [_ipython, _plain_shell]
+
+
+@command()
+@config_option
+@option('--plain', '-p',
+        is_flag=True,
+        default=False,
+        help='Use a plain shell instead of IPython if available.')
+def shell(config, plain):
+    """Run a Python interactive interpreter in the Flask app context."""
```

Comment:
Oh, you might want to make this a little shorter (and shunt the rest onto other lines) so `dxr --help` looks good. It truncates long (and even not-so-long) lines.